packages/stream-model-instance/src/model-instance-document.ts

```diff
@@ -218,18 +239,25 @@ export class ModelInstanceDocument<T = Record<string, any>> extends Stream {
    * @param opts - Additional options
    */
   async replace(content: T | null, opts: UpdateOpts = {}): Promise<void> {
-    opts = { ...DEFAULT_UPDATE_OPTS, ...opts }
+    const { shouldIndex, ...options } = { ...DEFAULT_UPDATE_OPTS, ...opts }
```

Comment:
I'm still pretty opposed to putting this into the `UpdateOpts`.  That seems like a huge abuse of that type.  If we're going to though we should at least actually add it to the type definition in StreamOpts.ts.